File: BE/services/TripScheduleService.js
import tripScheduleRepository from '../repositories/TripScheduleRepository.js';
import { TripSchedule } from '../models/TripSchedule.js';

/**
 * Service xử lý business logic cho lịch đi chơi
 */
class TripScheduleService {
  getAllTripSchedules() {
    return tripScheduleRepository.findAll();
  }

  getTripScheduleById(id) {
    return tripScheduleRepository.findById(id);
  }

  createTripSchedule(tripScheduleData) {
    const tripSchedule = new TripSchedule(tripScheduleData);
    return tripScheduleRepository.save(tripSchedule);
  }

  updateTripSchedule(id, tripScheduleData) {
    if (!tripScheduleRepository.existsById(id)) {
      throw new Error(`Không tìm thấy lịch đi chơi với ID: ${id}`);
    }
    const tripSchedule = new TripSchedule({ ...tripScheduleData, id });
    return tripScheduleRepository.save(tripSchedule);
  }

  deleteTripSchedule(id) {
    if (!tripScheduleRepository.existsById(id)) {
      throw new Error(`Không tìm thấy lịch đi chơi với ID: ${id}`);
    }
    tripScheduleRepository.deleteById(id);
  }
}

export default new TripScheduleService();

File: BE/src/main/java/com/food/app/service/TripScheduleService.java
package com.food.app.service;

import com.food.app.model.TripSchedule;
import com.food.app.repository.TripScheduleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

/**
 * Service xử lý business logic cho lịch đi chơi
 */
@Service
public class TripScheduleService {
    
    @Autowired
    private TripScheduleRepository tripScheduleRepository;
    
    public List<TripSchedule> getAllTripSchedules() {
        return tripScheduleRepository.findAll();
    }
    
    public Optional<TripSchedule> getTripScheduleById(Long id) {
        return tripScheduleRepository.findById(id);
    }
    
    public TripSchedule createTripSchedule(TripSchedule tripSchedule) {
        return tripScheduleRepository.save(tripSchedule);
    }
    
    public TripSchedule updateTripSchedule(Long id, TripSchedule tripSchedule) {
        if (!tripScheduleRepository.existsById(id)) {
            throw new RuntimeException("Không tìm thấy lịch đi chơi với ID: " + id);
        }
        tripSchedule.setId(id);
        return tripScheduleRepository.save(tripSchedule);
    }
    
    public void deleteTripSchedule(Long id) {
        if (!tripScheduleRepository.existsById(id)) {
            throw new RuntimeException("Không tìm thấy lịch đi chơi với ID: " + id);
        }
        tripScheduleRepository.deleteById(id);
    }
}


